SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift

```diff
@@ -288,6 +296,104 @@ var splunkRumInitializeCalledTime = Date()
         initialized = true
         print("SplunkRum.initialize() complete")
         return true
+    }
+
+    @discardableResult
+    @objc internal class func create(beaconUrl: String, rumAuth: String, options: SplunkRumOptions? = nil) -> Bool {
```

Comment:
should we just make the options non-optional here? This is only called from the builder correct? And that always passes non nil options.

Reply:
The ones that should be mandatory, in my opinion, are:

- beaconUrl || realm
- rumAuth

Rest could be optional. But I'd like to see realm coming first, to replicate the same behavior of our other RUM instrumentations.

Reply:
makes sense! I just think since the builder always calls into this function with a non-nil SplunkOptions object, it would clean things up here to not accept an optional. The user can of course chose to call `build` with as many builder configuration functions as they need :)

Reply:
Also, Android has a requirement on applicationName. Should iOS also introduce this requirement and add it to the init params?

Reply:
Not if the iOS distro already retrieves that name automatically.

Reply:
it does but it can fail. And that name might not be what the user wants to see on their dashboard. Basically, should iOS and Android be the same here?

Reply:
I think it's best if we replicate behavior as much as possible. +CC @dvernon-splunk

Reply:
pending a possible change here, the new commits look fine to me

Reply:
Added an optional appName parameter to the create function. Not providing an appName will fall back on old behavior.